### PR TITLE
 python312Packages.django-q2: init at 1.7.4; mailmanPackages.hyperkitty: 1.3.9 -> 1.3.12; mailmanPackages: use python 3.12; python3Packages.django-q: drop

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/servers/mail/mailman/python.nix [hunk 1]
```diff
@@ -20,13 +20,14 @@ lib.fix (self: python3.override {
...
-          hash = "sha256-WF3FFrnrBCphnvCjnD19Vf6BvbTfCaUsnN3g0Hvxqn0=";
+      # the redis python library only supports hiredis 3+ from version 5.1.0 onwards
+      hiredis = super.hiredis.overrideAttrs ({ src, ... }: let
+        version = "3.1.0";
+      in {
+        inherit version;
+        src = src.override {
+          rev = "refs/tags/v${version}";
+          hash = "sha256-ID5OJdARd2N2GYEpcYOpxenpZlhWnWr5fAClAgqEgGg=";
         };
```
Comment (getchoo) [inline pkgs/servers/mail/mailman/python.nix:31]:
```suggestion
      hiredis = super.hiredis.overrideAttrs (new: { src, ... }: {
        version = "3.1.0";
        src = src.override {
          rev = "refs/tags/v${new.version}";
          hash = "sha256-ID5OJdARd2N2GYEpcYOpxenpZlhWnWr5fAClAgqEgGg=";
        };
```

File: pkgs/development/python-modules/django-q2/default.nix [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    croniter
+    django-redis
+    # pyredis refuses to load with hiredis<3.0.0
+    (hiredis.overrideAttrs (old: rec {
+      version = "3.1.0";
+      src = old.src.override {
+        rev = "refs/tags/v${version}";
+        hash = "sha256-ID5OJdARd2N2GYEpcYOpxenpZlhWnWr5fAClAgqEgGg=";
+      };
+    }))
```
Comment (getchoo) [inline pkgs/development/python-modules/django-q2/default.nix:58]:
```suggestion
    (hiredis.overrideAttrs (new: old: {
      version = "3.1.0";
      src = old.src.override {
        rev = "refs/tags/v${new.version}";
        hash = "sha256-ID5OJdARd2N2GYEpcYOpxenpZlhWnWr5fAClAgqEgGg=";
      };
    }))
```

**Review by getchoo — COMMENTED** [review]
Some ways to avoid `rec`/plain `let ...; in`. Rest of the diff LGTM

**Review by getchoo — APPROVED** [review]

File: pkgs/servers/mail/mailman/hyperkitty.nix [hunk 3]
```diff
@@ -7,15 +7,15 @@
...
 
   disabled = pythonOlder "3.10";
 
   src = fetchurl {
     url = "https://gitlab.com/mailman/hyperkitty/-/releases/${version}/downloads/hyperkitty-${version}.tar.gz";
-    hash = "sha256-BfhCh4zZcfwoIfubW/+MUWXwh1yFOH/jpRdQdsj6lME=";
+    hash = "sha256-3rWCk37FvJ6pwdXYa/t2pNpCm2Dh/qb9aWTnxmfPFh0=";
   };
 
   nativeBuildInputs = [
```
Comment (dotlambda) [inline pkgs/servers/mail/mailman/hyperkitty.nix:21]:
```suggestion
  build-system = [
```

**Comment by SuperSandro2000** [comment]
Also took the opportunity to outright drop django-q

**Comment by SuperSandro2000** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 363754`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>mailman</li>
    <li>mailman-hyperkitty</li>
    <li>mailman-hyperkitty.dist</li>
    <li>mailman-web</li>
    <li>mailman-web.dist</li>
    <li>mailman.dist</li>
    <li>python311Packages.django-q2</li>
    <li>python311Packages.django-q2.dist</li>
    <li>python312Packages.django-q2</li>
    <li>python312Packages.django-q2.dist</li>
  </ul>
</details>

**Pull request merged**